packages/scan/src/core/monitor/performance-store.ts

```diff
@@ -0,0 +1,115 @@
+import { BoundedArray } from 'src/core/monitor/performance-utils';
+import {
+  InternalInteraction,
+  PerformanceInteraction,
+} from 'src/core/monitor/types';
+
+// forgot why start time was here ngl
+type Item = any;
+type UnSubscribe = () => void;
+type Callback = (item: Item) => void;
+type Updater = (state: BoundedArray<Item>) => BoundedArray<Item>;
+type ChanelName = string;
+
+type PerformanceEntryChannelsType = {
+  subscribe: (to: ChanelName, cb: Callback) => UnSubscribe;
+  publish: (
+    item: Item,
+    to: ChanelName,
+    dropFirst: boolean,
+    createIfNoChannel: boolean,
+  ) => void;
+  channels: Record<
+    ChanelName,
+    { callbacks: BoundedArray<Callback>; state: BoundedArray<Item> }
+  >;
+  getAvailableChannels: () => BoundedArray<string>;
+  updateChannelState: (
+    channel: ChanelName,
+    updater: Updater,
+    createIfNoChannel: boolean,
+  ) => void;
+};
```

Comment:
This is a little more strongly typed here.. and you can implement `PerformanceEntryChannelsType` 

```suggestion
export type ChannelItem = InternalInteraction | PerformanceInteraction;
export type UnSubscribe = () => void;
export type Callback<T> = (item: T) => void;
export type Updater<T> = (state: BoundedArray<T>) => BoundedArray<T>;
export type ChannelName = string;

interface Channel<T> {
  callbacks: BoundedArray<Callback<T>>;
  state: BoundedArray<T>;
}

interface PerformanceEntryChannelsType<T> {
  subscribe: (to: ChannelName, cb: Callback<T>, dropFirst?: boolean) => UnSubscribe;
  publish: (
    item: T,
    to: ChannelName,
    createIfNoChannel?: boolean
  ) => void;
  channels: Record<ChannelName, Channel<T>>;
  getAvailableChannels: () => BoundedArray<string>;
  updateChannelState: (
    channel: ChannelName,
    updater: Updater<T>,
    createIfNoChannel?: boolean
  ) => void;
  getChannelState: (channel: ChannelName) => BoundedArray<T>;
}
```